File: patterns/associated-items/rules-python/associated_items_naming.py
from openhab import rule, Registry, logger
from openhab.triggers import GroupStateChangeTrigger


@rule(triggers=[GroupStateChangeTrigger("gSensors")])
class ZugehoerigesItemUeberNamenskonventionFinden:
    def execute(self, module, input):
        event = input.get("event")
        if not event:
            return

        item_name = event.getItemName()

        # Status-Items selbst nicht weiterverarbeiten, sonst würde nach
        # "Sensor1_Status_Status" gesucht, das es nicht gibt
        if item_name.endswith("_Status"):
            return

        try:
            status_item = Registry.getItem(item_name + "_Status")
        except Exception:
            logger.warn("Kein zugehoeriges Status-Item fuer " + item_name + " gefunden")
            return

        # Den tatsaechlichen neuen Zustand des Sensors uebernehmen,
        # statt ihn fest auf ON zu setzen
        new_state = event.getItemState()
        status_item.postUpdate(new_state)
        logger.info("{} -> {} = {}".format(item_name, status_item.getName(), new_state))

File: patterns/associated-items/rules-js/associated-items-naming.js
const { rules, triggers, items } = require('openhab');

rules.JSRule({
  name: "Zugehoeriges Item ueber Namenskonvention finden",
  triggers: [triggers.ItemGroupStateChangeTrigger("gSensors")], // Angepasst
  execute: (event) => {
    if (event.itemName.endsWith("_Status")) { //[cite: 2]
      return;
    }
    const statusItem = items.getItem(event.itemName + "_Status"); //[cite: 2]
    statusItem.postUpdate("ON"); //[cite: 2]
  }
});
